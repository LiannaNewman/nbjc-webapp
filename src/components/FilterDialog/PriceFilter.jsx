import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

import { Chip, Typography } from '@material-ui/core';
import CheckIcon from '@material-ui/icons/Check';

const styles = {
  wrapper: {
    display: 'inline',
  },
  filter: {
    marginRight: 10,
    minWidth: 42,
    textAlign: 'center',
  },
};

const PriceFilter = ({ filters, classes, onFilterClick }) => (
  <div className={classes.wrapper}>
    {filters.map((filter) => (
      <Chip
        label={(
          <Typography color={filter.active ? 'inherit' : 'primary'} variant="body2">
            {filter.label}
          </Typography>
        )}
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...filter.active && { icon: <CheckIcon /> }}
        color={filter.active ? 'primary' : 'default'}
        className={classes.filter}
        onClick={() => onFilterClick(filter.value)}
        key={filter.value}
      />
    ))}
  </div>
);

PriceFilter.propTypes = {
  filters: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.number.isRequired,
      label: PropTypes.string.isRequierd,
      active: PropTypes.bool,
    }),
  ).isRequired,
  onFilterClick: PropTypes.func.isRequired,
  classes: PropTypes.shape({}).isRequired,
};

export default withStyles(styles)(PriceFilter);
