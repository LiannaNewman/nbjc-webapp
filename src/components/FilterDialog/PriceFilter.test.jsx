import React from 'react';
import { render } from '@testing-library/react';
import PriceFilter from './PriceFilter';

test('renders the BusinessCardComponent', () => {
  const active = 1;
  const priceFilters = [...Array(4)].map((_, i) => ({
    label: [...Array(i + 1)].map(() => '$').join(),
    value: i + 1,
    active: active === i + 1,
  }));
  const filterClick = jest.fn();
  const { asFragment } = render(<PriceFilter filters={priceFilters} onFilterClick={filterClick} />);
  expect(asFragment()).toMatchSnapshot();
});
