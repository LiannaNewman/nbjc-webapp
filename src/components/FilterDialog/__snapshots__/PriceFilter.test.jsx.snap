// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the BusinessCardComponent 1`] = `
<DocumentFragment>
  <div
    class="PriceFilter-wrapper-1"
  >
    <div
      class="MuiButtonBase-root MuiChip-root PriceFilter-filter-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiChip-icon MuiChip-iconColorPrimary"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
        />
      </svg>
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorInherit"
        >
          $
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root PriceFilter-filter-2 MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorPrimary"
        >
          $,$
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root PriceFilter-filter-2 MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorPrimary"
        >
          $,$,$
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root PriceFilter-filter-2 MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorPrimary"
        >
          $,$,$,$
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
